docs/tutorials/typescript/01-intro-typescript.md

```diff
@@ -0,0 +1,86 @@
+</head>
+
+:::note
+TypeScript/JavaScript support for WebAssembly is rapidly evolving.
+This page was last revised in September 2025.
+:::
+
+Kubewarden uses [Javy](https://github.com/bytecodealliance/javy) (a Bytecode Alliance project) to build WebAssembly binaries from JavaScript.
+
+> Javy takes your JavaScript code and executes it in a WebAssembly context. 
+>
+> It features an embedded QuickJS engine, compiled to WebAssembly, that can execute JavaScript. 
+>
+> The project provides both a CLI and a set of APIs for embedding and customizing the behavior when running JavaScript in WebAssembly.
+
+The Kubewarden project currently uses Javy for these reasons:
+
+- Mature JavaScript engine (QuickJS) compiled to WebAssembly
+- Support for [waPC](https://wapc.io) through custom host functions
+- Smaller binary sizes compared to other JavaScript-to-WebAssembly solutions
+- Active development and maintenance by the Bytecode Alliance
+
+## Javy limitations
+
+Javy runs JavaScript in a sandboxed WebAssembly environment with certain constraints:
+
+- **WASI environment only**: Access limited to stdin/stdout/stderr and explicitly provided host capabilities
+- **No Node.js APIs**: Standard Node.js modules like `fs`, `http`, or `crypto` are not available
+- **Limited standard library**: Only core JavaScript features and explicitly enabled APIs are accessible
+- **Single-threaded execution**: No support for Web Workers or multi-threading
+
+Despite these limitations, Javy provides sufficient capabilities for writing effective Kubewarden validation policies through the host capabilities system.
+
+## Tooling
+
+Writing Kubewarden policies requires:
+
+- **Node.js**: Version 18 or higher
+- **npm**: For dependency management
+- **TypeScript**: Recommended for type safety (optional)
+
+:::warning
+Ensure you're using Node.js 18 or higher. Older versions may not be compatible with the compilation toolchain.
+:::
+
+These TypeScript/JavaScript libraries are useful when writing a Kubewarden policy:
+
+- [Kubewarden JavaScript SDK](https://github.com/kubewarden/policy-sdk-js): Provides structures and functions reducing the amount of code necessary. It also provides test helpers and access to all host capabilities.
+- [Kubernetes TypeScript types](https://github.com/silverlyra/kubernetes-types): Provides TypeScript definitions for all Kubernetes resources, enabling type-safe policy development.
+
+The Kubewarden project provides a [template JavaScript/TypeScript policy project](https://github.com/kubewarden/js-policy-template) you can use to create Kubewarden policies.
+
+## Getting the toolchain
+
+The easiest way to get the complete toolchain is by using the Kubewarden JavaScript SDK, which includes the Javy compilation plugin:
+
+```bash
+npm install kubewarden-policy-sdk
+```
+
+The Javy plugin binary is automatically included and can be found at:
```

Comment:
Sorry, unrelated with the docs... Does it mean the binary plugin is shipped via npm? If so, how do we deal with operating systems and architectures?

Reply:
My fault, the plugin is actually a `.wasm` module, hence we don't need to build it for other OS/architectures.